src/crimson/osd/osd_operations/peering_event.cc

```diff
@@ -94,7 +96,40 @@ RemotePeeringEvent::ConnectionPipeline &RemotePeeringEvent::cp()
   return get_osd_priv(conn.get()).peering_request_conn_pipeline;
 }
 
-seastar::future<Ref<PG>> RemotePeeringEvent::get_pg() {
+void RemotePeeringEvent::on_pg_absent()
+{
+  if (auto& e = get_event().get_event();
+      e.dynamic_type() == MQuery::static_type()) {
+    const auto map_epoch =
+      shard_services.get_osdmap_service().get_map()->get_epoch();
+    const auto& q = static_cast<const MQuery&>(e);
+    const pg_info_t empty{spg_t{pgid.pgid, q.query.to}};
+    if (q.query.type == q.query.LOG ||
+	q.query.type == q.query.FULLLOG)  {
+      auto m = new MOSDPGLog{q.query.from, q.query.to,
```

Comment:
Is there a reason for not using `make_message`?

Reply:
```c++
struct BufferedRecoveryMessages {
  void send_osd_message(int target, Message *m) {
    message_map[target].push_back(m);
  }
```
i will update `BufferedRecoveryMessages` in a follow-up PR to accept a smart pointer.

Reply:
#30412